pandas/tests/reshape/test_melt.py

```diff
@@ -661,3 +661,36 @@ def test_col_substring_of_stubname(self):
                                  i=['node_id', 'A'],
                                  j='time')
         tm.assert_frame_equal(result, expected)
+
+    def test_melt_missing_columns(self):
+        # GH-23575
+        # This test is to ensure that pandas raises an error if melting is
+        # attempted with column names absent from the dataframe
+
+        # Generate data
+        df = pd.DataFrame(np.random.randn(5, 4), columns=list('abcd'))
+
+        # Try to melt with missing `value_vars` column name
+        with pytest.raises(KeyError, match="The following 'value_vars' are not"
```

Comment:
Typically we write this as

```
msg = (
    "The following ... "
    " ... "
    "the DataFrame"
)
with pytest.raises(KeyError, match=msg)
```

This is also a regex, so you can simplify to something like `msg="*'value_vars'.* 'C'"` (untested).

Reply:
Regex wasn't working and now when I try pytest with `match` equal to the exact output of the error message (as before) I get this frustrating error:

`
AssertionError: Pattern 'The following 'value_vars' are not present in the DataFrame: ['C']' not found in '"The following 'value_vars' are not present in the DataFrame: ['C']"'
`